### PR TITLE
Feature/cc welcome page basics

### LINKED ISSUE #12: WelcomePageView

### PR DESCRIPTION
**Pull request by carlymclift**
#### What's this PR do?
* Renders the Welcome Page with instructions, history, current stories, and published story components
* styles the Welcome page with basic styling 
* Tests that the correct things are rendered on the page 
* tests that when buttons are clicked, the necessary behavior occurs

#### Where should the reviewer start?
WelcomePage.js

#### How should this be manually tested?
 run `npm test` in the testing suite.
and 
Open up browser to see the changes on the main home page

#### Any background context you want to provide?
There is mock data currently for the prompts that will be replaced by api data

#### What are the relevant tickets?
closes #12 

#### Screenshots (if appropriate)

Screen Shot 2020-09-12 at 5 54 17 PM
Screen Shot 2020-09-12 at 5 54 24 PM
Screen Shot 2020-09-12 at 5 54 31 PM

### CODE REVIEW

File: src/components/GameHistory/GameHistory.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
+.game-history-box {
+  position: fixed;
+  z-index: 1;
```
Comment (Abdeboskey) [inline src/components/GameHistory/GameHistory.css:3]: Nice use of the z-index here. That modal pop up is clean!

File: src/components/GameHistory/GameHistory.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            <div className="drawing" style={{ backgroundImage: `url(${drawing3})` }}><p>1933</p></div>
+            <div className="drawing" style={{ backgroundImage: `url(${drawing4})` }}><p>1934</p></div>
+            <div className="drawing" style={{ backgroundImage: `url(${drawing5})` }}><p>1935</p></div>
+            <div className="drawing" style={{ backgroundImage: `url(${drawing6})` }}><p>2000</p></div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
```
Comment (Abdeboskey) [inline src/components/GameHistory/GameHistory.js:39]: This is a really great feature, I'm so glad you did this!

File: src/components/Instructions/Instructions.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      </div>
+      <button className="show-history-button" onClick={displayHistory}>History of Exquisite Corpse</button>
+      {showHistory ? <GameHistory displayHistory={displayHistory} /> : null}
+      <Link to='/story-setup' style={{ fontSize: 30 }}className="start-new-story-link">Start a new story</Link>
+      <h2>OR</h2>
+      <h3 className="text">Continue a story:</h3>
+    </section>
+  )
+}
+
```
Comment (Abdeboskey) [inline src/components/Instructions/Instructions.js:29]: I really appreciate how you've chunked these components into leetol bits.

File: src/components/Instructions/Instructions.test.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    const { getByRole } = render(
+      <MemoryRouter>
+        <Instructions 
+          displayHistory={mockFun}
+        />
+      </MemoryRouter>
+      )
+
+    fireEvent.click(getByRole('button', { name: 'History of Exquisite Corpse' }))
```
Comment (nickhartdev) [inline src/components/Instructions/Instructions.test.js:41]: Never though of using this shorthand, I like it

File: src/components/GameHistory/GameHistory.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import './GameHistory.css'
+import drawing1 from "../../assets/drawing1.png";
+import drawing2 from "../../assets/drawing2.png";
+import drawing3 from "../../assets/drawing3.png";
+import drawing4 from "../../assets/drawing4.png";
+import drawing5 from "../../assets/drawing5.png";
+import drawing6 from "../../assets/drawing6.png";
+
+const GameHistory = ( { displayHistory } ) => {
```
Comment (nickhartdev) [inline src/components/GameHistory/GameHistory.js:10]: Really liking this section! Thanks for doing this

File: src/components/Instructions/Instructions.test.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+         showHistory={true}
+        />
+      </MemoryRouter>
+    )
+
+    const hiddenHistory = screen.getByText('Exquisite Corpse, also known', { exact: false })
+
+    expect(hiddenHistory).toBeInTheDocument()
+
+  })
```
Comment (Abdeboskey) [inline src/components/Instructions/Instructions.test.js:59]: Way to get these tests happening from the start!

**Review by nickhartdev — APPROVED** [review]
Looks good! These changes are looking solid. Especially the history section, I really like that part

File: src/components/PromptCards/PromptCards.css [hunk 7]
```diff
@@ -0,0 +1,33 @@
+.PromptCards-sec {
+    display: flex;
+    flex-direction: row;
```
Comment (Abdeboskey) [inline src/components/PromptCards/PromptCards.css:3]: I think that flex containers default to the `row` flex-direction. this line may not be necessary!

File: src/components/PromptCards/PromptCards.css [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    color: rgb(70, 0, 128);
+    font-weight: 400;
+  }
+
+  .published-stories-text {
+    margin-left: 40%;
+    font-size: 30px;
+    color: rgb(70, 0, 128);
+    font-weight: 400;
+  }
```
Comment (Abdeboskey) [inline src/components/PromptCards/PromptCards.css:33]: Your styling game is on point.

File: src/components/PromptCards/PromptCards.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <div className="library-sec">
+      <div className="PromptCards-sec">
+          {stories}
+      </div>
+      <h3 className="published-stories-text" >View Published Stories</h3>
+    </div>
+  )
+}
+
```
Comment (Abdeboskey) [inline src/components/PromptCards/PromptCards.js:24]: This is perfect, definitely distinguished between in-progress and published more clearly.

File: src/components/WelcomePageView/WelcomePageView.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+      })
+     }
+
+    displayHistory = () => {
+      this.setState(prevState => {
+        return {
+            showHistory: !prevState.showHistory
+        }
+      })
+    }
```
Comment (Abdeboskey) [inline src/components/WelcomePageView/WelcomePageView.js:34]: Nice use of prevState here

**Review by Abdeboskey — APPROVED** [review]
Heck yeah Carly, this is awesome. Thank you!

File: src/components/GameHistory/GameHistory.js (hunk 5 above)
Comment (GreysonElkins) [reply src/components/GameHistory/GameHistory.js:10]: Yes! Such a nice touch!

File: src/components/PromptCards/PromptCards.css (hunk 7 above)
Comment (GreysonElkins) [reply src/components/PromptCards/PromptCards.css:3]: I also think that's the default

File: src/components/PromptCards/PromptCards.js (hunk 9 above)
Comment (GreysonElkins) [reply src/components/PromptCards/PromptCards.js:24]: How would we feel about just rendering in-progress stories from this page? Keep the options trim and singlular?

File: src/components/WelcomePageView/WelcomePageView.js (hunk 10 above)
Comment (GreysonElkins) [reply src/components/WelcomePageView/WelcomePageView.js:34]: !!!! I want to learn way more about whatever this is

File: src/components/WelcomePageView/WelcomePageView.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+                showHistory={this.state.showHistory}
+                displayHistory={this.displayHistory}
+              />
+            </div>
+            <div>
+              <PromptCards 
+                inProgressStories={this.state.inProgressStories}
+              />
+            </div>
+            <LibraryView />
```
Comment (GreysonElkins) [inline src/components/WelcomePageView/WelcomePageView.js:50]: I believe that you can replace `LibraryView` with something like this. 
```
 <Bookshelf
          stories={whateverStoriesHere}
          onClick={whateverFunctionHere}
        />
```
As of right now, LibraryView will only render "complete stories" from it's own lifecycle methods, and if you click the books, they may render a story view for the book you clicked (I think this functionality hasn't been implemented and will be coming with my next PR, however). 

If you render a Bookshelf via the method above, you'll beable to customize the selection of books you want to see and pass that selection as a prop / whatever method you pass into the onClick will pass the "story" object corresponding to the clicked book as an argument. So in my code it looks like:
 ```  
selectStoryToRead = (story) => {
    this.setState({ currentStory: story})
  }
```
but that method could easily be switched to something which changes the Route to an `editStoryView`.

**Review by GreysonElkins — APPROVED** [review]
@carlymclift I approve, and if you'd like go ahead and merge it! 
I left some notes about the `<LibraryView>` vs `<Bookshelf>` question; if you want to make that change for this Push I'm happy to pair with you on it / we can get that fix in later as well. Your call

_edit_: after looking closer I realized I misunderstood what we were talking about. I imagined you might want the bookshelf for the incomplete stories and didn't think we'd be rendering published stories on the home page. We come back around to that, but I'm gonna go ahead and merge this afterall since I was not at all en pointe a minute ago talking about the `<Bookshelf />`

Great job on this!

**Pull request merged**